File: src/stores/content.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { invoke } from '@tauri-apps/api/core'

export interface ContentItem {
  id: string
  account_id: string
  title: string
  body: string
  tags: string
  images: string
  scheduled_at: string | null
  status: string
  created_at: string
}

export const useContentStore = defineStore('content', () => {
  const queue = ref<ContentItem[]>([])
  const loading = ref(false)

  async function fetchQueue() {
    loading.value = true
    try {
      queue.value = await invoke<ContentItem[]>('get_content_queue')
    } catch (e) {
      console.error('Failed to fetch content queue:', e)
    } finally {
      loading.value = false
    }
  }

  async function resetStalePublishing() {
    const stale = queue.value.filter(i => i.status === 'publishing')
    for (const item of stale) {
      await updateStatus(item.id, 'pending')
    }
  }

  async function addContent(item: ContentItem) {
    try {
      await invoke('add_content', { item })
      await fetchQueue()
    } catch (e) {
      console.error('Failed to add content:', e)
      throw e
    }
  }

  async function updateContent(item: ContentItem) {
    try {
      await invoke('update_content', { item })
      await fetchQueue()
    } catch (e) {
      console.error('Failed to update content:', e)
      throw e
    }
  }

  async function updateStatus(id: string, status: string) {
    try {
      await invoke('update_content_status', { id, status })
      await fetchQueue()
    } catch (e) {
      console.error('Failed to update content status:', e)
      throw e
    }
  }

  async function deleteContent(id: string) {
    try {
      await invoke('delete_content', { id })
      await fetchQueue()
    } catch (e) {
      console.error('Failed to delete content:', e)
      throw e
    }
  }

  return { queue, loading, fetchQueue, resetStalePublishing, addContent, updateContent, updateStatus, deleteContent }
})

File: src-tauri/src/commands/content.rs
use crate::db::{ContentItem, Database};
use tauri::State;

#[tauri::command]
pub fn get_content_queue(db: State<'_, Database>) -> Result<Vec<ContentItem>, String> {
    db.get_content_queue().map_err(|e| e.to_string())
}

#[tauri::command]
pub fn get_content_by_id(db: State<'_, Database>, id: String) -> Result<Option<ContentItem>, String> {
    db.get_content_by_id(&id).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn add_content(db: State<'_, Database>, item: ContentItem) -> Result<(), String> {
    db.add_content(&item).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn update_content(db: State<'_, Database>, item: ContentItem) -> Result<(), String> {
    db.update_content(&item).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn update_content_status(db: State<'_, Database>, id: String, status: String) -> Result<(), String> {
    db.update_content_status(&id, &status).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn delete_content(db: State<'_, Database>, id: String) -> Result<(), String> {
    db.delete_content(&id).map_err(|e| e.to_string())
}
